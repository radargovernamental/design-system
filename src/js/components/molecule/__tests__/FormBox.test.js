import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import FormBox from '../FormBox';

configure({ adapter: new Adapter() });

describe('FormBox', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<FormBox>Lorem</FormBox>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with a child fieldset', () => {
    const wrapper = shallow(<FormBox>Lorem</FormBox>);
    expect(wrapper.is('fieldset')).toBe(true);
  });

  it('renders with children inside it', () => {
    const wrapper = shallow(<FormBox><div>Lorem <strong>Ipsum</strong></div></FormBox>);
    expect(wrapper.children().html()).toBe('<div>Lorem <strong>Ipsum</strong></div>');
  });
});
