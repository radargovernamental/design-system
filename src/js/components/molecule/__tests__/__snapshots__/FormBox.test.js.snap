// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBox renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormBox
    className=""
    isLoading={false}
>
    Lorem
</FormBox>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Lorem",
      "className": "undefined ",
      "disabled": false,
    },
    "ref": null,
    "rendered": "Lorem",
    "type": "fieldset",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem",
        "className": "undefined ",
        "disabled": false,
      },
      "ref": null,
      "rendered": "Lorem",
      "type": "fieldset",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
